packages/runed/src/lib/utilities/element-size/element-size.svelte.ts

```diff
@@ -22,50 +23,77 @@ export type ElementSizeOptions = ConfigurableWindow & {
  * @see {@link https://runed.dev/docs/utilities/element-size}
  */
 export class ElementSize {
-	#size = $state({
+	// no need to use `$state` here since we are using createSubscriber
+	#size = {
 		width: 0,
 		height: 0,
-	});
+	};
 
-	constructor(node: MaybeElementGetter, options: ElementSizeOptions = { box: "border-box" }) {
-		const window = options.window ?? defaultWindow;
+	#options: ElementSizeOptions;
 
-		this.#size = {
-			width: options.initialSize?.width ?? 0,
-			height: options.initialSize?.height ?? 0,
-		};
+	#node: MaybeElementGetter;
+
+	#window: ElementSizeOptions["window"];
 
-		$effect(() => {
-			if (!window) return;
-			const node$ = get(node);
-			if (!node$) return;
+	// we use a derived here to extract the width so that if the width doesn't change we don't get a state update
+	// which we would get if we would just use a getter since the version of the subscriber will be changing
+	#width = $derived.by(() => {
+		this.#subscribe?.();
+		return this.#size.width;
+	});
 
-			const observer = new window.ResizeObserver((entries) => {
+	// we use a derived here to extract the height so that if the height doesn't change we don't get a state update
+	// which we would get if we would just use a getter since the version of the subscriber will be changing
+	#height = $derived.by(() => {
+		this.#subscribe?.();
+		return this.#size.height;
+	});
+
+	// we need to use a derived here because the class will be created before the node is bound to the ref
+	#subscribe = $derived.by(() => {
+		const node$ = get(this.#node);
+		if (!node$) return;
```

Comment:
Can't you get node inside the createSubscriber instead, to avoid using the derived? I'd also maybe add a node that is a derived with `extract`.

Reply:
Nope because you need to register the dependency on the node, the function within createSubscriber is untracked (if I'm not mistaken, or it's run later).

I did add a derived with extract at one point but that's just extra indirection no?

Reply:
yeah it is, would be worthy more in case this was possible